google/gax/api_callable.py

```diff
@@ -417,7 +439,9 @@ def base_caller(api_call, _, *args):
 
     def inner(request, options=None):
         """Invoke with the actual settings."""
-        this_settings = settings.merge(options)
+        this_options = _merge_options_metadata(options, settings)
```

Comment:
Please check the previous comment for more details about the merging logic.

Yes, another place, where the merging can be put is settings.merge() method itself. I chose to put it in separate method because: 1) putting it in the merge method looks too dangerous, 2) the _ClientSettings class does not know about "special" 'metadata' field, while the api_callable.py already does some magic with that field in construct_settings (so I decided to not proliferate the knowledge about "magical metadata" to multiple files, and keep it in one place - api_callable.py).